### PR TITLE
Update pulumi terraform provider to 0.10.0.

### PR DESCRIPTION
**Pull request by markfickett**
I can build the Python SDK and see new resources. (In my mian project I have some proto library dependency issues, but if I use `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python` it works fine.)

All the indirect go.mod entries are autogenerated by `go mod tidy`.

### CODE REVIEW

File: provider/go.mod [hunk 1]
```diff
@@ -1,14 +1,217 @@
...
+	google.golang.org/protobuf v1.28.1 // indirect
+	gopkg.in/AlecAivazis/survey.v1 v1.8.9-0.20200217094205-6773bdf39b7f // indirect
+	gopkg.in/square/go-jose.v2 v2.6.0 // indirect
+	gopkg.in/src-d/go-billy.v4 v4.3.2 // indirect
+	gopkg.in/src-d/go-git.v4 v4.13.1 // indirect
+	gopkg.in/warnings.v0 v0.1.2 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+	lukechampine.com/frand v1.4.2 // indirect
+	sourcegraph.com/sourcegraph/appdash v0.0.0-20211028080628-e2786a622600 // indirect
```
Comment (antifuchs) [inline provider/go.mod:216]: Oh wow, this seems to have blown up. Usually if the list gets too large, that may indicate a problem but I assume it's fine if you can build & use it!

**Review by antifuchs — APPROVED** [review]
This looks great! Unfortunately, due to our corp requirements need the CLA signed in order to get it merged - hope that's fine.

For the python protobuf issues, I can highly recommend the `"upb"` mode instead of "python" - it's far faster and doesn't require this incredibly messy CPP library to be installed.

File: provider/go.mod (hunk 1 above)
Comment (markfickett) [reply provider/go.mod:216]: Maybe -- I'm definitely half guessing about all of this, so if you have time to at least check that this still builds properly and works for you that would be good.

Comment (antifuchs) [reply provider/go.mod:216]: My policy has been that if `go mod tidy` is happy and it builds, that's the correct set of deps to include. (:

**Review by antifuchs — CHANGES_REQUESTED** [review]
Oops, I forgot one thing, can you run `make go-sdk` and commit the resulting files? The go SDK is the only one that has vc-tracked source files because that's how go rolls.

**Comment by markfickett** [comment]
Updated with the `go` sdk.

I've agreed to the CLA. When I revisit the page it says 'you have agreed' even though it says pending here, is there a way to re-trigger that? (Same applies to the other PR.)

**Comment by markfickett** [comment]
Figured out the CLA issue, had the wrong committer email on my git commits so it didn't track, fixed that & also found the recheck link on the CLA message here. Should be all set!

**Comment by antifuchs** [comment]
Think it needed a push to this branch,, the universal "please retry" button for github (-:

I think we're good now, thank you!!

**Review by antifuchs — APPROVED** [review]

**Pull request merged**